### PR TITLE
[cxx-interop] SWIFT_RETURNS_INDEPENDENT_VALUE implies immortal lifetime

### PR DESCRIPTION
**Pull request by Xazax-hun**
The semantics of returning independent value already matches what immortal lifetimes are within Swift. This patch makes sure this annotation works as expected with non-escapable types.

rdar://137671642

### CODE REVIEW

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test

**Review by egorzhdan — APPROVED** [review]
This seems very reasonable, LGTM!

File: lib/ClangImporter/ImportDecl.cpp [hunk 1]
```diff
@@ -3854,6 +3854,18 @@ namespace {
       if (decl->getTemplatedKind() == clang::FunctionDecl::TK_FunctionTemplate)
         return;
 
+      SmallVector<LifetimeDependenceInfo, 1> lifetimeDependencies;
+      LifetimeDependenceInfo immortalLifetime(nullptr, nullptr, 0,
+                                              /*isImmortal*/ true);
+      if (const auto *funDecl = dyn_cast<FuncDecl>(result))
```
[the remark this answers is not in the document]
Comment (Xazax-hun) [reply lib/ClangImporter/ImportDecl.cpp:3860]: Ah, my bad! I actually wanted to use it. Missing a check here 😅

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test windows

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test windows

**Pull request merged**